drivers/gnss/quectel_lcx6g.c

```diff
@@ -0,0 +1,757 @@
+		break;
+	}
+}
+
+static int quectel_lcx6g_get_navigation_mode(const struct device *dev,
+					     enum gnss_navigation_mode *mode)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)dev->data;
+	k_spinlock_key_t key;
+	int ret;
+
+	key = k_spin_lock(&data->lock);
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				     "PAIR081");
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_script_chat.request_size = ret;
+
+	strncpy(data->dynamic_match_buf, "$PAIR081,", sizeof(data->dynamic_match_buf));
+	data->dynamic_match.match_size = sizeof("$PAIR081,") - 1;
+	data->dynamic_match.callback = quectel_lcx6g_get_navigation_mode_callback;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	data->dynamic_match.callback = NULL;
+	*mode = data->navigation_mode_response;
+	k_spin_unlock(&data->lock, key);
+	return 0;
+}
+
+static int quectel_lcx6g_set_enabled_systems(const struct device *dev, gnss_systems_t systems)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)dev->data;
+	gnss_systems_t supported_systems;
+	k_spinlock_key_t key;
+	int ret;
+
+	supported_systems = (GNSS_SYSTEM_GPS | GNSS_SYSTEM_GLONASS | GNSS_SYSTEM_GALILEO |
+			     GNSS_SYSTEM_BEIDOU | GNSS_SYSTEM_QZSS | GNSS_SYSTEM_SBAS);
+
+	if ((~supported_systems) & systems) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->lock);
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				     "PAIR066,%u,%u,%u,%u,%u,0",
+				     (0 < (systems & GNSS_SYSTEM_GPS)),
+				     (0 < (systems & GNSS_SYSTEM_GLONASS)),
+				     (0 < (systems & GNSS_SYSTEM_GALILEO)),
+				     (0 < (systems & GNSS_SYSTEM_BEIDOU)),
+				     (0 < (systems & GNSS_SYSTEM_QZSS)));
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_script_chat.request_size = ret;
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_match_buf, sizeof(data->dynamic_match_buf),
+				     "PAIR001,066,0");
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_match.match_size = ret;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				     "PAIR410,%u", (0 < (systems & GNSS_SYSTEM_SBAS)));
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_script_chat.request_size = ret;
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_match_buf, sizeof(data->dynamic_match_buf),
+				     "PAIR001,410,0");
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_match.match_size = ret;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	k_spin_unlock(&data->lock, key);
+	return ret;
```

Comment:
I mean, you did all the others as

```c
if (ret < 0) {
	k_spin_unlock(&data->lock, key);I
	return ret;
}
```
may as well do this one as well and `return 0`. Same in the other functions above, a bit weird to change pattern on the last one.

Reply:
updated functions to use the same pattern for the last action